docs/architecture/modernize-desktop/01-why-modern-desktop-applications.md

```diff
@@ -0,0 +1,169 @@
+
+## Desktop applications nowadays
+
+Before the raise of the Internet, desktop applications were the main approach to build software systems. You could choose any programming language like Cobol, Fortran, Visual Basic, C++, etc., but you ended up creating some sort of desktop application, from small tools to complex distributed architectures.
+
+Then, Internet technologies started shocking the development world and winning over more and more engineers with advantages like easy deployment and simplified distribution processes. The fact that once Web application was deployed to production all users got automatic updates made a huge impact on the software agility.
+
+However, the Internet infrastructure, underlying protocols and standards like HTTP and HTML were not design for building complex applications. In fact, the major development effort back then was aiming just one goal: to give Web applications same capabilities that desktop applications have, such as fast data input, state management and so on.
+
+Even though Web and Mobile applications have grown at an incredible pace, for certain tasks desktop applications still hold the number one place in terms of efficiency and performance. That explains why there are millions of developers who are building their projects with WPF and WinForms and the amount of those applications is constantly growing.
+
+Here are some reasons for choosing desktop applications in your development:
+
+- Desktop apps have better interaction with user's PC.
+- The performance of desktop applications for complex calculations is much higher than performance of web applications.
+- Running custom logic on the client side is possible but much harder with a web application.
+- Using multithreading is easier and more efficient in a desktop application.
+- The learning curve for designing User Interfaces is smother and in case of WinForms is completely intuitive with drag-and-drop experience of the Windows Forms designer.
+- It is easy to start coding and testing your algorithms without the need to set up a server infrastructure or to care about connectivity problems, firewalls and browser compatibility.
+- Debugging is powerful as compared to web debugging.
+- Access to hardware devices like camera, Bluetooth or card readers is easy.
+- And since the technology have been around for a long time there is a huge human expertise and knowledge database available on developing desktop applications.
+
+So, as you can see, developing for desktop is great for many reasons. The technology is mature and time tested, the development cycle is fast, the debugging is powerful and arguably, desktop apps have less complexity and easier to get started with.
+
+Microsoft offered a variety of UI desktop technologies through the years from Win32 introduced in 1995 and to UWP released in 2016.
+
+![Microsoft desktop technologies](./media/1/microsoft-desktop-technologies.png)
```

Comment:
instead of 1, the folder name should match the article name like 
./media/why-modern-desktop-applications/microsoft-desktop-technologies.png

Reply:
Fixed